api/java/Dockerfile

```diff
@@ -7,7 +7,9 @@ COPY pom.xml .
 COPY src src
 COPY .env .env
 
-RUN mvn clean package
+ENV SIMULATE_MONGO true
+RUN mvn clean install 
+ENV SIMULATE_MONGO false

```

Comment:
You're overwriting the above variable two lines above.

Reply:
Ok, we're back to the issue of what the automated tests are supposed to be. If they are not to test the queuing and status setting functionality, what are they supposed to be doing?

Reply:
unit tests are not integration tests.  The unit test should only test the details within the function, not the dependencies.  It requires mocking those dependencies.

Reply:
The SIMULATE_MONGO = true implies that no real database is used. It will connect to an embedded DB. Does that clarify things?

Reply:
i.e. a "mocked" database

Reply:
Sorry for the confusion. I mentioned in the review working on getting the embedded DB to work. Thought you realized what it was meant for.